suite/tests/client-interface/drreg-test.c

```diff
@@ -494,6 +535,56 @@ GLOBAL_LABEL(FUNCNAME:)
 #endif
         END_FUNC(FUNCNAME)
 #undef FUNCNAME
+
+#define FUNCNAME test_asm_faultE
+        DECLARE_FUNC_SEH(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+#ifdef X86
+#ifdef X64
+        PUSH_CALLEE_SAVED_REGS()
+        sub      REG_XSP, FRAME_PADDING /* align */
+        END_PROLOG
+
+          /* XXX i#3312: Temporarily disable test until bug has been fixed. */
+#if 0
+        jmp      test10
+        /* Test 10: fault test restore of non-public DR slot used by mangling,
+         * when rip-rel address is forced to be in register. Making sure drreg ignores
+         * restoring this slot.
+         */
+     test10:
+        mov      PTRSZ [REG_XSP], REG_XAX
+        sub      REG_XSP, 8
+        mov      TEST_REG_ASM, DRREG_TEST_10_ASM
+        mov      TEST_REG_ASM, DRREG_TEST_10_ASM
+        nop
+        mov      REG_XAX, DRREG_TEST_11_ASM
+        /* There's a good chance that this will be too far away for a
+         * signed 32-bit rip-rel offset, which is what we want, in order
+         * to get the address mangled into register REG_XAX.
+         */
+        mov      REG_XAX, PTRSZ SYMREF(-0x7fffffff) /* crash */
```

Comment:
How about just making this an add that reads xax too and then there is no need for a special option?

Reply:
You're right. Done.